contracts-periphery/src/test/UmbraBatchSend.t.sol

```diff
@@ -0,0 +1,186 @@
+            .checked_write(1000*1e18);
+    }
+
+    function testBatchSendEth() public {
+
+        uint alicePrevBal = alice.balance;
+        uint bobPrevBal = bob.balance;
+
+        uint amount = 1 ether;
+        uint amount2 = 2 ether;
+        uint256 toll = 0;        
+
+        sendEth.push(UmbraBatchSend.SendEth(payable(alice), amount, test, test));
+        sendEth.push(UmbraBatchSend.SendEth(payable(bob), amount2, test, test));
+        router.batchSendEth{value: amount + amount2}(toll, sendEth);
+
+        assertEq(alice.balance, alicePrevBal + amount);
+        assertEq(bob.balance, bobPrevBal + amount2);
+
+    }
+
+    function testFuzz_BatchSendEth(uint8 amount, uint8 amount2, bytes32 testBytes) public {
+
+        uint alicePrevBal = alice.balance;
+        uint bobPrevBal = bob.balance;
+        vm.assume(amount > 0 && amount < type(uint8).max/2);
+        vm.assume(amount2 > 0 && amount2 < type(uint8).max/2);
+        
+
+        uint256 toll = 0;
+
+        sendEth.push(UmbraBatchSend.SendEth(payable(alice), amount, testBytes, testBytes));
+        sendEth.push(UmbraBatchSend.SendEth(payable(bob), amount2, testBytes, testBytes));
+
+        uint totalAmount = amount + amount2;
+
+        router.batchSendEth{value: totalAmount}(toll, sendEth);
+
+        assertEq(alice.balance, alicePrevBal + amount);
+        assertEq(bob.balance, bobPrevBal + amount2);
+
+    }
+
+    function testBatchSendTokens() public {
+
+        assertTrue(token.balanceOf(address(this)) > 0, "caller's dai balance is zero");
+
+        uint256 toll = 0;
+        uint umbraPrevBal = token.balanceOf(umbra);
+
+        sendToken.push(UmbraBatchSend.SendToken(alice, Dai, 1000, test, test));
+        sendToken.push(UmbraBatchSend.SendToken(bob, Dai, 500, test, test));
+        token.approve(address(router), 1500);
+
+        router.batchSendTokens{value: toll}(toll, sendToken);
+        assertEq(token.balanceOf(umbra), umbraPrevBal + 1500);
+
+    }
+
+    function testFuzz_BatchSendTokens(uint8 amount, uint8 amount2) public {
+
+        assertTrue(token.balanceOf(address(this)) > 0, "caller's dai balance is zero");
+
+        vm.assume(amount < type(uint8).max/2);
+        vm.assume(amount2 < type(uint8).max/2);
+
+        uint256 toll = 0;
+        uint umbraPrevBal = token.balanceOf(umbra);
+
+        sendToken.push(UmbraBatchSend.SendToken(alice, Dai, amount, test, test));
+        sendToken.push(UmbraBatchSend.SendToken(bob, Dai, amount2, test, test));
+        token.approve(address(router), amount + amount2);
+
+        router.batchSendTokens{value: toll}(toll, sendToken);
+        assertEq(token.balanceOf(umbra), umbraPrevBal + amount + amount2);
+
+    }
+
+    function testBatchSend() public {
+
+        uint alicePrevBal = alice.balance;
+        uint bobPrevBal = bob.balance;
+
+        uint amount = 1 ether;
+        uint amount2 = 2 ether;
+        uint total = amount + amount2;
+        uint256 toll = 0;        
```

Comment:
the tests have a lot of repetitive variable declarations that can be pulled into the `setUp` method to deduplicate